### DISCUSSION TITLE
Tailwind CSS does not generate all 3rd part components classes

### DISCUSSION THREAD

**Discussion by Garfield550** [tailwindlabs/tailwindcss#8426] [Help] [answered]
**What version of Tailwind CSS are you using?**

^3.0.24

**What build tool (or framework if it abstracts the build tool) are you using?**

```json
"dependencies": {
  "react": "^18.1.0",
  "react-dom": "^18.1.0",
  "react-tabs": "^5.1.0"
},
"devDependencies": {
  "@vitejs/plugin-react": "^1.3.2",
  "autoprefixer": "^10.4.7",
  "postcss": "^8.4.14",
  "tailwindcss": "^3.0.24",
  "typescript": "^4.6.4",
  "vite": "^2.9.9",
  "vite-tsconfig-paths": "^3.4.1"
}
```

**What version of Node.js are you using?**

v16.15.0

**What browser are you using?**

Microsoft Edge Version 101.0.1210.53 (Official build) (arm64)

**What operating system are you using?**

macOS

**Reproduction URL**

https://github.com/Garfield550/tailwind-demo

**Describe your issue**

Class `react-tabs__tab-panel--selected` not generated, `TabPanel` not display correctly.

<img width="1065" alt="image" src="https://user-images.githubusercontent.com/3471836/170065566-b27e47d6-dc43-47f7-843e-437e8baf0562.png">

**Comment by adamwathan** [comment, marked as answer]
Hey! The issue here is that the `react-tabs` library doesn't actually include the string `react-tabs__tab-panel--selected` anywhere in its source code, so Tailwind doesn't see it and doesn't generate it. The best solution in your situation is to _not_ put those styles in the actual `components` layer so they aren't only included when used, and instead just include them all the time by putting them in the same spot in the file as the components layer but not in the actual layer:

```css
@tailwind base;
@tailwind components;

/* react-tabs */
.react-tabs {
  @apply [-webkit-tap-highlight-color:transparent] bg-white;
}
.react-tabs__tab-list {
  @apply m-0 p-0 flex flex-row;
}
.react-tabs__tab {
  @apply flex flex-row items-center justify-center box-border cursor-pointer list-none focus:outline-none
      w-[170px] h-[50px] mx-[15px] border-[3px] border-solid border-transparent bg-transparent
      text-center font-bold text-base text-[#333333] leading-none [letter-spacing:0.05em] uppercase;
}
.react-tabs__tab--selected {
  @apply [text-shadow:1px_-1px_14px_#46FFE6] border-[#46FFE6];
}
.react-tabs__tab--disabled {
  @apply text-[#4F4F4F] cursor-default;
}
.react-tabs__tab--warned:after {
  @apply content-['_'] inline-block w-1.5 h-1.5 rounded-full bg-[#FF906D];
}
/* .react-tabs__tab:focus {} */
/* .react-tabs__tab:focus:after {} */
.react-tabs__tab-panel {
  @apply hidden;
}
.react-tabs__tab-panel--selected {
  @apply block;
}

@tailwind utilities;
```

**Reply by Garfield550** [reply]
Thanks for your answer.
